panic after incorrect `--secret=id` parameter

### Issue Description

podman panics when providing an incomplete `--secret` parameter

### Steps to reproduce the issue

> podman --version
podman version 5.8.1
> # In a directory with a valid `Dockerfile`
> podman build --secret=id=,src .
panic: runtime error: index out of range [1] with length 1
...

### Describe the results you received

panic: runtime error: index out of range [1] with length 1

goroutine 13 [running]:
github.com/containers/buildah/pkg/parse.Secrets({0xc0006c2710, 0x1, 0x0?})
        github.com/containers/buildah/pkg/parse/parse.go:1091 +0x568
github.com/containers/buildah/imagebuildah.newExecutor(_, {_, _}, {_, _}, {{0x0, 0x0}, {0xc000116b80, 0x1c}, 0x0, ...}, ...)
        github.com/containers/buildah/imagebuildah/executor.go:200 +0x727
github.com/containers/buildah/imagebuildah.buildDockerfilesOnce({_, _}, {_, _}, _, {_, _}, {{0x0, 0x0}, {0xc000116b80, ...}, ...}, ...)
        github.com/containers/buildah/imagebuildah/build.go:429 +0x967
github.com/containers/buildah/imagebuildah.BuildDockerfiles.func2()
        github.com/containers/buildah/imagebuildah/build.go:269 +0x4cd
github.com/hashicorp/go-multierror.(*Group).Go.func1()
        github.com/hashicorp/go-multierror/group.go:23 +0x5f
created by github.com/hashicorp/go-multierror.(*Group).Go in goroutine 1
        github.com/hashicorp/go-multierror/group.go:20 +0x79

### Describe the results you expected

I did not expect a panic. Maybe an error if the parameter is mandatory.

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.43.0
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: conmon-2.2.1-2.fc43.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.2.1, commit: '
  cpuUtilization:
    idlePercent: 99.84
    systemPercent: 0.1
    userPercent: 0.06
  cpus: 8
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: container
    version: "43"
  eventLogger: file
  freeLocks: 2048
  hostname: 75457642090f
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 1
      size: 999
    - container_id: 1000
      host_id: 1001
      size: 64535
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 1
      size: 999
    - container_id: 1000
      host_id: 1001
      size: 64535
  kernel: 6.6.87.2-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 6089797632
  memTotal: 8172826624
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    defaultNetwork: podman
    dns:
      package: aardvark-dns-1.17.0-1.fc43.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.17.0
    package: netavark-1.17.2-1.fc43.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.17.2
  ociRuntime:
    name: crun
    package: crun-1.25.1-1.fc43.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.25.1
      commit: 156ae065d4a322d149c7307034f98d9637aa92a2
      rundir: /tmp/storage-run-1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20260120.g386b5f5-1.fc43.x86_64
    version: |
      pasta 0^20260120.g386b5f5-1.fc43.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /tmp/storage-run-1000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 2147483648
  swapTotal: 2147483648
  uptime: 2h 59m 11.00s (Approximately 0.08 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /home/podman/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/podman/.local/share/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 29818691584
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /tmp/storage-run-1000/containers
  transientStore: false
  volumePath: /home/podman/.local/share/containers/storage/volumes
version:
  APIVersion: 5.8.1
  BuildOrigin: Fedora Project
  Built: 1773187200
  BuiltTime: Wed Mar 11 00:00:00 2026
  GitCommit: c6077f645788743258a1a749f8005b4fb3cbe533
  GoVersion: go1.25.7 X:nodwarf5
  Os: linux
  OsArch: linux/amd64
  Version: 5.8.1
```

### Podman in a container

Yes

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

I found it using podman version 4.9.3 on WSL with Ubuntu 24.04.4 LTS.
I tested it with the latest podman image to get the latest version. (using podman run on the above environment.)

### Additional information

The sting provided with `id=` does not matter.

--- a/pkg/parse/parse.go
+++ b/pkg/parse/parse.go
@@ -1319,20 +1319,23 @@ func Secrets(secrets []string) (map[string]define.Secret, error) {
 		tokens := strings.Split(secret, ",")
 		var id, src, typ string
 		for _, val := range tokens {
-			kv := strings.SplitN(val, "=", 2)
-			switch kv[0] {
+			key, value, ok := strings.Cut(val, "=")
+			if !ok {
+				return nil, errInvalidSecretSyntax
+			}
+			switch key {
 			case "id":
-				id = kv[1]
+				id = value
 			case "src":
-				src = kv[1]
+				src = value
 			case "env":
-				src = kv[1]
+				src = value
 				typ = "env"
 			case "type":
-				if kv[1] != "file" && kv[1] != "env" {
+				if value != "file" && value != "env" {
 					return nil, errors.New("invalid secret type, must be file or env")
 				}
-				typ = kv[1]
+				typ = value
 			default:
 				return nil, errInvalidSecretSyntax
 			}

--- a/pkg/parse/parse_test.go
+++ b/pkg/parse/parse_test.go
@@ -2,6 +2,7 @@ package parse //nolint:revive,nolintlint
 
 import (
 	"fmt"
+	"os"
 	"runtime"
 	"testing"
 
@@ -296,6 +297,64 @@ func TestGetBuildOutput(t *testing.T) {
 			result, err := GetBuildOutput(testCase.input)
 			require.NoErrorf(t, err, "expected to be able to parse %q", testCase.input)
 			assert.Equal(t, testCase.output, result)
+		})
+	}
+}
+
+func TestSecrets(t *testing.T) {
+	errorTests := []struct {
+		name  string
+		input string
+	}{
+		{"bare-src-no-equals", "id=,src"},
+		{"bare-id-no-equals", "id"},
+		{"known-key-without-value", "id=mysecret,src"},
+		{"empty-id", "id="},
+		{"unknown-key", "id=mysecret,bogus=x"},
+	}
+	for _, tc := range errorTests {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := Secrets([]string{tc.input})
+			assert.Error(t, err, "expected error for input %q", tc.input)
+		})
+	}
+
+	tmpfile, err := os.CreateTemp(t.TempDir(), "secret")
+	require.NoError(t, err)
+	_, err = tmpfile.WriteString("supersecret")
+	require.NoError(t, err)
+	tmpfile.Close()
+
+	t.Setenv("TEST_SECRET_ENV", "envsecret")
+
+	validFile := tmpfile.Name()
+
+	successTests := []struct {
+		name     string
+		input    string
+		expected define.Secret
+	}{
+		{
+			"id-and-file-src",
+			"id=mysecret,src=" + validFile,
+			define.Secret{ID: "mysecret", Source: validFile, SourceType: "file"},
+		},
+		{
+			"id-and-env",
+			"id=TEST_SECRET_ENV,env=TEST_SECRET_ENV",
+			define.Secret{ID: "TEST_SECRET_ENV", Source: "TEST_SECRET_ENV", SourceType: "env"},
+		},
+		{
+			"id-only-env-fallback",
+			"id=TEST_SECRET_ENV",
+			define.Secret{ID: "TEST_SECRET_ENV", Source: "TEST_SECRET_ENV", SourceType: "env"},
+		},
+	}
+	for _, tc := range successTests {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := Secrets([]string{tc.input})
+			require.NoError(t, err)
+			assert.Equal(t, tc.expected, result[tc.expected.ID])
 		})
 	}
 }

--- a/tests/bud.bats
+++ b/tests/bud.bats
@@ -7074,6 +7074,20 @@ _EOF
   expect_output --substring "ENVDATA"
 }
 
+@test "bud with malformed --secret flag should fail not panic" {
+  run_buildah 125 build --secret=id=,src $WITH_POLICY_JSON -f $BUDFILES/run-mounts/Dockerfile.secret $BUDFILES/run-mounts
+  expect_output --substring "incorrect secret flag format"
+
+  run_buildah 125 build --secret=id=mysecret,src $WITH_POLICY_JSON -f $BUDFILES/run-mounts/Dockerfile.secret $BUDFILES/run-mounts
+  expect_output --substring "incorrect secret flag format"
+
+  run_buildah 125 build --secret=src $WITH_POLICY_JSON -f $BUDFILES/run-mounts/Dockerfile.secret $BUDFILES/run-mounts
+  expect_output --substring "incorrect secret flag format"
+
+  run_buildah 125 build --secret=id $WITH_POLICY_JSON -f $BUDFILES/run-mounts/Dockerfile.secret $BUDFILES/run-mounts
+  expect_output --substring "incorrect secret flag format"
+}
+
 @test "bud-multiple-platform-values" {
   skip "FIXME: #4396 - this test is broken, and is failing gating tests"
   outputlist=testlist

